modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/EditSegmentsEntryDisplayContext.java

```diff
@@ -377,6 +378,16 @@ public long getSegmentsEntryId() {
 		return _segmentsEntryId;
 	}
 
+	public String getSource() throws PortalException {
+		SegmentsEntry segmentsEntry = getSegmentsEntry();
+
+		if (segmentsEntry != null) {
+			return segmentsEntry.getSource();
+		}
+
+		return ParamUtil.getString(_request, "source", SegmentsConstants.SOURCE_DEFAULT);
+	}
```

Comment:
@epgarcia @darquesdev I had to add the source value to the display context. Can one of you guys also check if this is okay and if anything needs to be added/changed? Thanks!

Reply:
Hi @thektan,

Looks OK to me. Only, in the view I'd change the name of the property for consistency. Because everything is for segments, I would simply call it "source", instead of segmentSource (in the same way we use only "contributors"). 

Thanks